src/params.rs

```diff
@@ -114,21 +135,20 @@ impl<PRFH: Hasher, MSGH: Hasher> Params<PRFH, MSGH> {
     pub fn msg_hash_and_compute_checksum(&self, msg: &[u8]) -> Vec<u8> {
         let mut hasher = MSGH::new();
         let mut msg_buf = vec![0u8; MSGH::size()];
-        let mut hashed_msg = vec![0u8; self.m as usize];
+        let mut hashed_msg = vec![0u8; self.m];
         hasher.write(msg.to_vec());
         hasher.sum(&mut msg_buf);
-        hashed_msg[0..self.m as usize].clone_from_slice(&msg_buf[0..self.m as usize]);
+        hashed_msg[0..self.m].clone_from_slice(&msg_buf[0..self.m]);
         hashed_msg.append(&mut checksum(&hashed_msg));
         hashed_msg
     }
 
     pub fn compute_ladders(
-        &mut self,
+        &self,
```

Comment:
since this is under `&self` for now, we should use `&self` for `sign` and `public_key` also

Reply:
updated